File: examples/js/structs/src/address.ts
import { LHStruct, LHField } from 'littlehorse-client'
import { VariableType } from 'littlehorse-client/proto'

@LHStruct({ name: 'address' })
export class Address {
  @LHField(VariableType.INT)
  houseNumber!: number

  @LHField(VariableType.STR)
  street!: string

  @LHField(VariableType.STR)
  city!: string

  @LHField(VariableType.STR)
  planet!: string

  @LHField(VariableType.INT)
  zipCode!: number

  constructor(houseNumber?: number, street?: string, city?: string, planet?: string, zipCode?: number) {
    if (houseNumber !== undefined) this.houseNumber = houseNumber
    if (street !== undefined) this.street = street
    if (city !== undefined) this.city = city
    if (planet !== undefined) this.planet = planet
    if (zipCode !== undefined) this.zipCode = zipCode
  }

  toString(): string {
    return `${this.houseNumber} ${this.street}, ${this.city}, ${this.planet} ${this.zipCode}`
  }
}

File: examples/go/structdef/structs/address.go
package structs

import (
	"fmt"

	"github.com/littlehorse-enterprises/littlehorse/sdk-go/littlehorse"
)

const AddressStructDefName string = "address"
const AddressStructDefDescription string = "Address represents a physical address."

// Address represents a physical address.
type Address struct {
	HouseNumber int    `json:"houseNumber"`
	Street      string `json:"street"`
	City        string `json:"city"`
	Planet      string `json:"planet"`
	ZipCode     int    `json:"zipCode"`
}

func (Address) LHStructDef() littlehorse.LHStructDefInfo {
	return littlehorse.LHStructDefInfo{Name: AddressStructDefName, Description: AddressStructDefDescription}
}

func (a Address) String() string {
	return fmt.Sprintf("%d %s, %s, %s %d", a.HouseNumber, a.Street, a.City, a.Planet, a.ZipCode)
}
